### PR TITLE
Updated gitmigrate command to a) work, b) be friendlier. Also, fixed docopt option parsing.

### LINKED ISSUE #4025: The gitmigrate management command isn't working
After failing with #4024, I tried running `gitmigrate` directly, but it failed as well (see below). I'm already working on trying some things for a fix.

    jamalex@hummingbird:~$ kalite manage gitmigrate
    CommandError: Must specify a path to migrate the Git installed version of KA Lite from

    jamalex@hummingbird:~$ kalite manage gitmigrate /home/jamalex/fle/ka-lite
    -------------------------------------------------------------------
    Attempting to copy Database file from previous installation
    -------------------------------------------------------------------

    IOError: [Errno 2] No such file or directory: '/usr/share/kalite/kalite/database/data.sqlite'

Also, on a UX note, it's hard to know what arguments to use, since it doesn't tell you about the `args[0]` option in the help, and the `--path` option is well-hidden:

    jamalex@hummingbird:~$ kalite manage help gitmigrate
    Usage: kalite manage gitmigrate [options] 

    Options:
      -v VERBOSITY, --verbosity=VERBOSITY
                            Verbosity level; 0=minimal output, 1=normal output,
                            2=verbose output, 3=very verbose output
      --settings=SETTINGS   The Python path to a settings module, e.g.
                            "myproject.settings.main". If this isn't provided, the
                            DJANGO_SETTINGS_MODULE environment variable will be
                            used.
      --pythonpath=PYTHONPATH
                            A directory to add to the Python path, e.g.
                            "/home/djangoprojects/myproject".
      --traceback           Print traceback on exception
      --auto-pdb            Break into PDB on an uncaught exception
      --path=PATH           Git Install Path
      --version             show program's version number and exit
      -h, --help            show this help message and exit

### PR DESCRIPTION
**Pull request by jamalex**
The command was trying to put stuff in `KALITE_DIR` instead of `DEFAULT_DATABASE_PATH` and `CONTENT_ROOT`, which prevented it from working at all.

The improvements here also present more options to the user, allowing them to select an alternate location to load videos from, see how big the files there are, and choose whether to copy or move. More granular progress is then displayed during the transfer.

### CODE REVIEW

File: kalitectl.py [hunk 1]
```diff
@@ -663,12 +663,17 @@ def docopt(doc, argv=None, help=True, version=None, options_first=False):  # @Re
...
-        )
+        
+        # count the number of arguments that were not Django options, and then properly read in Django
+        # options from sys.argv (we can't get them from "DJANGO_OPTIONS", since that doesn't catch 
+        # long-form options like "--git-migrate")
+        kalite_command_arg_count = len(collected)
+        if "DJANGO_OPTIONS" in result:
+            kalite_command_arg_count -= 1
+        result['DJANGO_OPTIONS'] = sys.argv[kalite_command_arg_count+1:]
+
```
Comment (jamalex) [inline kalitectl.py]: @benjaoming The old code here didn't work when mixing argument types for Django options (long-form-args like "--git-migrate" were not caught in "collected", but left in "__left", and the indexing counts were off).

Let me know if you see any issues with the way I've structured it now. It just ignores any stuff matched by `DJANGO_OPTIONS` and then pulls all the Django options in from `sys.argv`.

**Comment by jamalex** [comment]
Also added in a fix for https://github.com/learningequality/ka-lite/issues/4024 (which includes a general fix to docopt parsing of Django options).

**Comment by jamalex** [comment]
If anyone knows whether it's possible to ensure the server isn't running (or to stop it) from within the management command, that might make this process safer. In fact, it might be good for us to be doing that with the `setup` command in general?

**Comment by jamalex** [comment]
Some failing tests; I need to fix the docopts stuff further still; doing so now.

**Comment by jamalex** [comment]
Remaining errors look to be unrelated (timeouts in a couple behave tests), so this should be good to go (following review).

**Comment by benjaoming** [comment]
@jamalex the fix provided fixes a one-off error, but you will have other problems if you add more "--long-option value" arguments. Intentionally, you should have triggered this (see beginning of kalitectl.py)

```python
# We do not understand --option value, only --option=value.
# Match all patterns of "--option value" and fail if they exist
__validate_cmd_options = re.compile(r"--[^\s-]+\s+[^-\s][^-\s]+")
if __validate_cmd_options.search(" ".join(sys.argv[1:])):
    sys.stderr.write("Please only use --option=value patterns. The option parser gets confused if you do otherwise.\n")
    sys.exit(1)
```

**Comment by benjaoming** [comment]
I will see if I can find a solution for this and get back.

**Comment by benjaoming** [comment]
It was the dash in "--git-migrate" that made it to not fail :/

**Comment by benjaoming** [comment]
Opened up #4038 instead

**Comment by jamalex** [comment]
> It was the dash in "--git-migrate" that made it to not fail :/

Ah, I tried renaming the argument to `gitmigrate` and I see the parse error now.

In the release notes, we currently don't tell people to use `=`. I'll fix that too.
http://field.learningequality.org/develop/docs/installguide/release_notes.html#id1

**Comment by jamalex** [comment]
Doing the full docopt reversion in this PR, and fixing the docs here.

**Comment by rtibbles** [comment]
As I am technically assigned to this, it looks reasonable to me. If @benjaoming thinks the option parsing is right, then I would say we're good to merge.

**Comment by jamalex** [comment]
Option parsing has been reverted, so it's not included. Merging.

**Pull request merged**